src/components/semantic/props/CheckboxDocProp.tsx

```diff
@@ -9,18 +9,31 @@ type CheckboxDocProps<K extends keyof D, D> =
     prop: K;
     label: string;
     checkedIfUndefined?: boolean;
+    disabled?: boolean;
 };
 
 export function CheckboxDocProp<K extends keyof D, D extends { [Key in K]?: boolean }>({
     doc,
     update,
     prop,
     label,
-    checkedIfUndefined
+    checkedIfUndefined,
+    disabled,
 }: CheckboxDocProps<K, D>) {
-    return <Label className={styles.checkboxLabel}>
+    useEffect(()=> {
+        if (disabled && doc[prop]) {
+            update({
+                ...doc,
+                [prop]: false,
```

Comment:
Not sure I agree with the function of this `useEffect` conceptually -- we won't be able to make checkboxes that are _set_ by default if we add this (i.e. when disabled, prop is true). Perhaps we could reuse `checkedIfUndefined`, like we do below?

```suggestion
                [prop]: checkedIfUndefined ?? false,
```